components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -1206,6 +1206,108 @@ public void removeSubscription(APIIdentifier identifier, int applicationId)
 		}
 	}
 
+	/**
+	 * Moves subscriptions of one app to another app
+	 *
+	 * @param fromIdentifier subscriptions of this app
+	 * @param toIdentifier   will be moved into this app
+	 * @return number of subscriptions moved
+	 * @throws AppManagementException
+	 */
+	public int moveSubscriptions(APIIdentifier fromIdentifier, APIIdentifier toIdentifier)
+			throws AppManagementException {
+
+		Connection conn = null;
+		ResultSet results = null;
+		PreparedStatement ps = null;
+		int fromAppId = -1;
+		int toAppId = -1;
+		int count;
+
+		try {
+
+			String getAppIdQuery = "SELECT APP_ID FROM APM_APP " +
+					"WHERE APP_PROVIDER=? AND APP_NAME=? AND APP_VERSION=?";
+
+			try {
+				conn = APIMgtDBUtil.getConnection();
+				ps = conn.prepareStatement(getAppIdQuery);
+			} catch (SQLException e) {
+				handleException("Failed to create database connection ", e);
+			}
+
+			try {
+				ps.setString(1, fromIdentifier.getProviderName());
+				ps.setString(2, fromIdentifier.getApiName());
+				ps.setString(3, fromIdentifier.getVersion());
+				results = ps.executeQuery();
+				if (results.next()) {
+					fromAppId = results.getInt(1);
+				}
+			} catch (SQLException e) {
+				APIMgtDBUtil.closeAllConnections(ps, conn, results);
```

Comment:
The most appropriate place to clean up all the resources is a finally block. For instance, this doesn't do any cleaning up of certain resources when the functionality gets invoked without producing in any exception.